lib/excheck/formatter.ex

```diff
@@ -13,8 +13,13 @@ defmodule ExCheck.Formatter do
 
   @doc false
   def handle_event(event = {:suite_finished, _run_us, _load_us}, config) do
-    updated_test_count = config.tests_counter.test + ExCheck.IOServer.total_tests
-    new_cfg = %{config | tests_counter: %{test: updated_test_count}}
+    updated_test_count =
```

Comment:
IMHO this would be more idiomatic:

```
# in handle_event
updated_tests_counter = update_tests_counter(config.tests_counter, ExCheck.IOServer.total_tests)

# then, below:
defp update_tests_counter(count, increment) when is_integer(count) do
  count + increment
end
defp update_tests_counter(counter, increment) when is_map(counter) do
  %{counter | test: counter.test + increment}
end
```

_Ideally_ you'd find out if tests_counter is a struct and pattern match on that struct.